luanti-org/luanti
# Basic LAN Discovery Implementation

This pr implements proller's lan implementation, which is adapted for minetest so that local servers can be discovered by clients on the same network in the server list.

This is for the feature request #10208 

## To do

This PR is a Work in Progress 

- [x] Fix crash when exiting non dedicated local game.
- [x] Fix crash when minetest is shutting down.
- [x] Remove unnecessary code.
- [ ] Decide on a setting name for the serverlist_lan setting.
- [x] Replace minetest with lunati
- [ ] Maintainability Changes / Cleanup
- [ ] General Cleanup

## How to test

Currently not finished but you can test it if you want.  

All you should have to do is compile minetest for your platform, then start a server and open a new client and look in the serverlist.
